EIPS/eip-7229.md

```diff
@@ -0,0 +1,195 @@
+
+### Storage slot of logic address optimization
+
+To further optimize the minimal upgradeable proxy by controlling the slot value for the logic address within the range of 255(inclusive), you can use the following opcode to reduce gas consumption:
+
+```
+# store logic address to slot of proxy contract
+PUSH1 <slot>           [slot]
+PUSH20 <logicAddress>  [logicAddress slot]
+DUP2                   [slot logicAddress slot]
+SSTORE                 [slot]          => storage(slot => logicAddress)
+
+# return deployedCode
+PUSH1 0x9              [0x9 slot]
+PUSH1 0x30             [0x30 0x9 slot]
+PUSH0                  [00 0x30 0x9 slot]
+CODECOPY               [slot]          ==> memory(0x00~0x8: 0x30~0x54(deployedCode1stPart))
+PUSH1 0xf8             [0xf8 slot]
+SHL                    [slotAfterShl]
+PUSH1 0x9              [0x9 slotAfterShl]    
+MSTORE                 []              ==> memory(0x9: slotAfterShl(deployedCode2ndPart))
+PUSH1 0x10             [0x10]
+PUSH1 0x39             [0x39 0x10]
+PUSH1 0xa              [0xa 0x39 0x10]     
+CODECOPY               []              ==> memory(0xa~0x38: 0x39~0x64(deployedCode3rdPart))
+PUSH1 0x1a             [0x1a]
+PUSH0                  [00 0x1a]
+RETURN
+
+# proxy contract (deployedcode)
+CALLDATASIZE        [calldatasize] 
+PUSH0               [00 calldatasize]
+PUSH0               [00 00 calldatasize]
+CALLDATACOPY        []     ==> memory(00~(calldatasize-1) => codedata)
+PUSH0               [00]
+PUSH0               [00 00]
+CALLDATASIZE        [calldatasize 00 00]
+PUSH0               [00 calldatasize 00 00]
+PUSH1               [slot 00 calldatasize 00 00] 
+SLOAD               [logicAddress 00 calldatasize 00 00]
+GAS                 [gas logicAddress 00 calldatasize 00 00]
+DELEGATECALL        [result]
+RETURNDATASIZE      [returnDataSize result]
+PUSH0               [00 returnDataSize result]
+PUSH0               [00 00 returnDataSize result]
+RETURNDATACOPY      [result] => memory(00~(RETURNDATASIZE - 1) => RETURNDATA)
+RETURNDATASIZE      [returnDataSize result] 
+PUSH0               [00 returnDataSize result] 
+DUP3                [result 00 returnDataSize result]
+PUSH1 0x18          [0x18 result 00 returnDataSize result]
+JUMPI				[00 returnDataSize result]
+REVERT              [result]
+JUMPDEST            [00 returnDataSize result]
+RETURN              [result]
+```
+
+The bytecode generated by the above opcodes is as follows `60xx73yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy8155600960305f3960f81b60095260106039600a39601a5ff3365f5f375f5f365f60545af43d5f5f3e3d5f82601857fd5bf3`, replace `xx` to a slot of 1byte and replace `yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy` to a address of 20bytes before deploying contract 
+
+## Test Cases
+
+Test cases are available at [https://github.com/xiaobaiskill/minimal-upgradable-proxy/tree/main/test](https://github.com/xiaobaiskill/minimal-upgradable-proxy/tree/main/test)
+
+
+## Security Considerations
+
+None.
+
+## References
```

Comment:
see [EIP-1](https://eips.ethereum.org/EIPS/eip-1) for the name of authorized sections, maybe could be `Reference Implementation` but then it has to come before `Security Considerations`